### PR TITLE
Add spreadCalculationMethod to InterestRatePayout

### PR DESCRIPTION
**Pull request by PayalKhanna**
Add spreadCalculationMethod to InterestRatePayout

### CODE REVIEW

File: rosetta-source/src/main/rosetta/product-asset-enum.rosetta [hunk 1]
```diff
@@ -132,3 +132,7 @@ enum FPVFinalPriceElectionFallbackEnum: <"Specifies the fallback provisions in r
 
 	FPVClose <"In respect of the Early Final Valuation Date, the provisions for FPV Close shall apply.">
 	FPVHedgeExecution <"In respect of the Early Final Valuation Date, the provisions for FPV Hedge Execution shall apply.">
+
+enum SpreadCalculationMethodEnum: <"Method by which asset swap spread is calculated, par-par or proceeds.">
```
Comment (iansloyan) [inline rosetta-source/src/main/rosetta/product-asset-enum.rosetta:136]: is there a definition of par-par or proceeds we can reference here? can we at least put them in quotation marks? "par-par" and "proceeds" 
Also is this only applicable to Asset Swaps or would we ever indicate a SpreadCalculationMethod for other products? Could this Enum expand in time to include other types of calculations? If so let's use asset swap as example. Consider description as follows:
<"Method by which a spread is calculated. E.g., "par-par" or "proceeds" are methods by which an asset swap spread is calculated typically.">

File: rosetta-source/src/main/rosetta/product-asset-type.rosetta [hunk 2]
```diff
@@ -95,6 +95,7 @@ type InterestRatePayout extends PayoutBase: <" A class to specify all of the ter
 	bondReference BondReference (0..1) <"Reference to a bond underlier to represent an asset swap or Condition Precedent Bond.">
 	fixedAmount calculation (0..1) <"Fixed Amount Calculation">
 	floatingAmount calculation (0..1) <"Floating Amount Calculation">
+    spreadCalculationMethod SpreadCalculationMethodEnum (0..1) <"Method by which asset swap spread is calculated, par-par or proceeds.">
```
Comment (iansloyan) [inline rosetta-source/src/main/rosetta/product-asset-type.rosetta:98]: Consider description as follows:
<"Method by which a spread is calculated. E.g., "par-par" or "proceeds" are methods by which an asset swap spread is calculated typically.">

**Review by iansloyan — CHANGES_REQUESTED** [review]
Please consider making the enum and type general and future proof for additional spread calc methods by adjusting the commentary - thanks

File: rosetta-source/src/main/rosetta/product-asset-enum.rosetta (hunk 1 above)
Comment (ankitkulkarni1) [reply rosetta-source/src/main/rosetta/product-asset-enum.rosetta]: I modified the description to be more generic.

The enum values which are derived from FpML model do not contain any description.

File: rosetta-source/src/main/rosetta/product-asset-type.rosetta (hunk 2 above)
Comment (ankitkulkarni1) [reply rosetta-source/src/main/rosetta/product-asset-type.rosetta]: I modified the description to be more generic. Thanks for the suggestion!

File: rosetta-source/src/main/rosetta/product-asset-enum.rosetta (hunk 1 above)
Comment (ankitkulkarni1) [reply rosetta-source/src/main/rosetta/product-asset-enum.rosetta]: @iansloyan can you please review again? Thank you!

**Comment by tomhealey-icma** [comment]
Please respond to comments or withdraw this PR.

**Comment by ankitkulkarni1** [comment]
@PayalKhanna do you know why the pipeline/checks failed for this MR after my latest commit?

**Comment by ankitkulkarni1** [comment]
This PR can be closed. I have created another one off version 5.x.x.

**Pull request closed without merging**